lib/exprotobuf.ex

```diff
@@ -46,9 +48,61 @@ defmodule Protobuf do
         end
     end
 
-    config |> parse(__CALLER__) |> Builder.define(config)
+    with {:ok, record_path} <- compile(:proto, opts[:from], opts[:other]),
+         [ok: erl_module]   <- compile(:bytecode, record_path),
+         config             <- %Config{config | erl_module: erl_module},
+      do: config |> parse(__CALLER__) |> Builder.define(config)
+  end
+
+  def compile(:proto, path, options) when is_binary(path) do
+    with {:ok, paths} <- extract_paths(path),
+         true         <- File.exists?(paths["fullpath"]) do
+      compile(:proto, paths, options)
+    else
+      :error -> {:error, "Something went wrong!"}
```

Comment:
Might be better to be a little more specific here, presumably the error is that we failed to extract a path right? Can we return an error that says which path failed to be extracted?